### DISCUSSION TITLE
Creality K1 Series

### DISCUSSION THREAD

**Discussion by 3DCoded** [3DCoded/3MS#3] [General Help]
Questions/Answers for using the 3MS with a Creality K1 Max

**Comment by 3DCoded** [comment]
The 3MS is not tied to the K1 Max's electronics. It runs on a separate board. See the [Instructions](https://3dcoded.github.io/3MS/instructions/).

**Comment by Jager-f** [comment]
I understand that it works on its own board. How does the printer and 3ms work synchronize if there is no physical connection between them?

**Comment by 3DCoded** [comment]
The 3MS's board is controlled by the same Klipper that your printer is controlled by and configured as another MCU. This way the 3MS macros can control them and Klipper can synchronize/desync the 3MS with your extruder.

**Reply by Jager-f** [reply]
The printer and 3ms are connected via USB?

**Reply by 3DCoded** [reply]
It looks like the K1 Max uses a Sonic Pad. In that case, the 3MS is connected to the Sonic Pad via USB.

**Reply by Jager-f** [reply]
In my case, the printer already has klipper. RPI seems superfluous

**Reply by 3DCoded** [reply]
Sorry. I'm not very familiar with the K1 Max. If it already has Klipper in it (not running on an external computer), then the 3MS should be connected via USB directly to the printer.

**Reply by 3DCoded** [reply]
@Jager-f Let me know if you have any more questions about setting up the 3MS on your printer.

**Reply by Jager-f** [reply]
K1 max has a small root partition, 3MS is not installed in it. On this printer, all add-ons are usually installed in /usr/data. I cloned 3MS into this folder. Need to change paths in config files. Need your help

**Reply by 3DCoded** [reply]
I'd be happy to help. Which paths need changing? Are you getting any specific errors?

**Reply by 3DCoded** [reply]
I've added experimental support for the K1 in the install script. Updated installation instructions are available [here](https://3dcoded.github.io/3MS/configuration/install/#install-script).

**Reply by brentdeacon23** [reply]
So am I correct in understanding (Say for example with the SKR Mini E3) once I have the steppers motors, extruders and the board assembled, this plugs directly into the K1's usb port? Or is firmware required on the board itself?

**Reply by 3DCoded** [reply]
Yes. The board (in this case an SKR) plugs directly into the K1's USB port. You will need to flash firmware to the SKR.

**Comment by pvilbig** [comment]
I am actually having an issue where the K!Max doesn't have enough to clone the repository. any idea on how to fix that? I took out all extra things but it only gets to 88% and errors out. Thanks in advance

**Reply by 3DCoded** [reply]
The entire repository of the 3MS is ~100MB (including the documentation). Most installations only need the `main` branch. Can you try running this command to clone the repository instead?

```
git clone -b main --single-branch https://github.com/3DCoded/3MS
```

That should reduce the size to less than 1MB.

**EDIT:**
This workaround is now in the documentation.

**Comment by pvilbig** [comment]
Wow, thank you so much. I really appreciate that and I see you even updated the site. You’re awesome. I really appreciate the work you did. I bought the 3-D chameleon and it’s been a pain in the ass so thank you so much!Sincerely,Paul Andrew Vilbig Jr.On Sep 14, 2024, at 9:39 AM, Christopher Mattar ***@***.***> wrote:﻿
The entire repository of the 3MS is ~100MB (including the documentation). Most installations only need the main branch. Can you try running this command to clone the repository instead?
git clone -b main --single-branch https://github.com/3DCoded/3MS

That should reduce the size to less than 1MB.

**Comment by pvilbig** [comment]
The only issue I have now is for some reason the variables folder is unable to be found.Sincerely,Paul Andrew Vilbig Jr.On Sep 14, 2024, at 9:39 AM, Christopher Mattar ***@***.***> wrote:﻿
The entire repository of the 3MS is ~100MB (including the documentation). Most installations only need the main branch. Can you try running this command to clone the repository instead?
git clone -b main --single-branch https://github.com/3DCoded/3MS

That should reduce the size to less than 1MB.

**Reply by 3DCoded** [reply]
Can you please put the exact error message you're facing here?

**Reply by 3DCoded** [reply]
Can you try changing this section in `3ms/main.cfg`?

```
[save_variables]
filename: /usr/data/variables.cfg
```

**Comment by pvilbig** [comment]
Sincerely,Paul Andrew Vilbig Jr.On Sep 14, 2024, at 11:19 AM, Christopher Mattar ***@***.***> wrote:﻿
Can you please put the exact error message you're facing here?

**Comment by pvilbig** [comment]
Ok so I was able to get past that but now the dynamic macros aren’t playing nicely lolSincerely,Paul Andrew Vilbig Jr.On Sep 14, 2024, at 11:25 AM, Christopher Mattar ***@***.***> wrote:﻿
Can you try changing this section in 3ms/main.cfg?
[save_variables]
filename: /usr/data/variables.cfg

**Reply by 3DCoded** [reply]
Did you run the K1-specific install script?

```
python3 install.py --path /usr/data/printer_data/config/3ms
```

**Reply by 3DCoded** [reply]
Also, can you confirm that the `macros.cfg` file is located in `/usr/data/printer_data/config/3ms`?

**Comment by pvilbig** [comment]
Yes I did the other script wouldn’t even runSincerely,Paul Andrew Vilbig Jr.On Sep 14, 2024, at 12:01 PM, Christopher Mattar ***@***.***> wrote:﻿
Did you run the K1-specific install script?
python3 install.py --path /usr/data/printer_data/config/3ms

**Comment by pvilbig** [comment]
Yes that is there as wellSincerely,Paul Andrew Vilbig Jr.On Sep 14, 2024, at 12:02 PM, Christopher Mattar ***@***.***> wrote:﻿
Also, can you confirm that the macros.cfg file is located in /usr/data/printer_data/config/3ms?

**Comment by 3DCoded** [comment]
What **EXACT** error is DynamicMacros giving you? The full error text and/or a screenshot will really help me debug this.

**Comment by pvilbig** [comment]
Sincerely,Paul Andrew Vilbig Jr.On Sep 14, 2024, at 12:12 PM, Christopher Mattar ***@***.***> wrote:﻿
What error is DynamicMacros giving you?

**Reply by 3DCoded** [reply]
If there was a screenshot there, it didn't come through. Can you try replying through Github, not email?

**Reply by pvilbig** [reply]
![image](https://github.com/user-attachments/assets/d50097af-d96c-42bf-99d2-c8c033320c5f)

that’s the error I get

**Reply by 3DCoded** [reply]
Thank you!
Did you install DynamicMacros using the instructions [here](https://3dcoded.github.io/DynamicMacros/setup/)?

You may need to use this command to clone the repo instead:

```
git clone -b main --single-branch https://github.com/3DCoded/DynamicMacros
```

If the `install.sh` script doesn't work there, try `install-k1.sh` (you may need to run a `git pull` to get it).

Thank you for your patience as I debug this. K1 series printers use a non-standard Klipper filesystem setup, so I'm doing all the research I can to figure out where files should be on the K1 series.

**Reply by pvilbig** [reply]
Sorry. I’m still learning the stuff. How would I do the git pull?

**Reply by pvilbig** [reply]
And not a problem! You did the hard work! I appreciate your patience with me! Lol

**Reply by 3DCoded** [reply]
To run the git pull, run these commands:

```
cd ~/DynamicMacros
git pull
```

If you just cloned the repository now, you shouldn't need to run the `git pull` command.

**Reply by pvilbig** [reply]
Unfortunately it is still throwing up the same code.

**Reply by 3DCoded** [reply]
Is there a `dynamicmacros.py` file located in `/usr/share/klipper/klippy/extras`? Also, did you restart klipper from the command line (not from the UI) with `sudo service klipper restart`?

**Reply by pvilbig** [reply]
Yes it is but it is in the folder /root/DynamicMacros

**Reply by 3DCoded** [reply]
Can you manually copy the file into the `extras` folder using this command:

```
cd ~/DynamicMacros
cp dynamicmacros.py /usr/share/klipper/klippy/extras/dynamicmacros.py
```

It seems like the K1 series printers use a custom version of Klipper by Creality, which can lead to unexpected issues. I'll try my best to work around those issues, but you may have to root your printer to use the 3MS, DynamicMacros, and many other Klipper plugins.

**Reply by pvilbig** [reply]
Ok so that is done but now I get this…
![image](https://github.com/user-attachments/assets/143b6fb9-2d2a-4cd8-9f60-d01ae9724795)

**Reply by pvilbig** [reply]
Before going too crazy should I try to reroot everything so this way it’s a clean install?



here is also the error code log

AttributeError: 'GCodeDispatch' object has no attribute '_build_status_commands'
[INFO] 2024-09-15 04:28:49,751 [root] [webhooks:set_client_info:202] webhooks client 1977988336: New connection
[INFO] 2024-09-15 04:28:49,755 [root] [webhooks:set_client_info:202] webhooks client 1977988336: Client info {'program': 'Moonraker', 'version': 'v0.9.3-0-g71f9e67'}
[INFO] 2024-09-15 04:28:49,781 [root] [webhooks:_handle_query:530] _handle_query objects/subscribe:{'webhooks': None}
[INFO] 2024-09-15 04:28:49,782 [root] [webhooks:_handle_query:552] _handle_query before complete.wait
[INFO] 2024-09-15 04:28:49,783 [root] [reactor:_dispatch_loop:340] _dispatch_loop current nice = -20
[INFO] 2024-09-15 04:28:49,784 [root] [reactor:_dispatch_loop:342] _dispatch_loop new nice = -20
[INFO] 2024-09-15 04:28:49,785 [root] [webhooks:_handle_query:554] _handle_query after complete.wait:{'eventtime': 4790.944601832, 'status': {'webhooks': {'state': 'error', 'state_message': 'Internal error during connect: \'GCodeDispatch\' object has no attribute \'_build_status_commands\'\n\nOnce the underlying issue is corrected, use the "RESTART"\ncommand to reload the config and restart the host software.\nPrinter is halted\n'}}}
[INFO] 2024-09-15 04:28:58,408 [root] [webhooks:set_client_info:202] webhooks client 1977638840: New connection

**Reply by 3DCoded** [reply]
> Before going too crazy should I try to reroot everything so this way it’s a clean install?

The error message you're facing wouldn't happen on standard Klipper. I would recommend rerooting everything.

**Reply by pvilbig** [reply]
I did and I am still getting the same error unfortunately

**Reply by 3DCoded** [reply]
Hmm... It seems like a rooted K1 Series printer still runs Creality's custom Klipper. I've added a workaround for DynamicMacros. To install it:

1. Use the git pull instructions I mentioned earlier
2. There should be a `dynamicmacros_k1.py` file now
3. Rerun the `install-k1.sh` script
4. Restart Klipper (`sudo service klipper restart`)

**Reply by pvilbig** [reply]
Ok I think I got closer lol but have this which I don’t think is too bad, but yeah lol.
![image](https://github.com/user-attachments/assets/9ab4c6e2-c0de-4533-8e7e-e9d9382c9987)

**Reply by 3DCoded** [reply]
That is closer. I've updated the code to try to fix this. To install it, use the previous instructions:

1. Use the git pull instructions I mentioned earlier
2. Rerun the `install-k1.sh` script
3. Restart Klipper (`sudo service klipper restart`)

**Comment by pvilbig** [comment]
Alright! Seems to have worked now I’m just trying to install the KlipperScreen fork but after running the git clone it doesn’t find the install file

**Reply by 3DCoded** [reply]
The installation script should be loacted in `scripts/KlipperScreen-install.sh` (in the `KlipperScreen` folder).

**Reply by pvilbig** [reply]
Ok got that but now this error pops up in the Klipper screen fork
![image](https://github.com/user-attachments/assets/a04abf8c-78eb-4f76-ac90-d5c1e30aa4d6)
I checked the lines of the code. Nothing seemed crazy but then again I’m still new to this lol

**Reply by 3DCoded** [reply]
That's strange. It seems like KlipperScreen is unable to find the `MMMS_SETTINGS` macro. Does running the `GET_MMMS_SETTINGS` macro in Fluidd/Mainsail work?

**Reply by pvilbig** [reply]
No the command is unknown.

**Reply by 3DCoded** [reply]
Is this line present in the `3ms/main.cfg` file (in the same folder as your `printer.cfg`)?

```
[include ./settings.cfg]
```

**Reply by 3DCoded** [reply]
The entire `3ms` folder should be in the same folder as your `printer.cfg`.

**Reply by pvilbig** [reply]
Yes all those things are good

**Reply by 3DCoded** [reply]
Can you share the contents of `3ms/main.cfg` and `3ms/settings.cfg`? Also, is `3ms/main.cfg` included in your `printer.cfg` with:

```
[include 3ms/main.cfg]
```

**Reply by ENF3471** [reply]
Hello,  I just rooted my K1 max.  This is my first time really using Klipper.  Is there any step by step video instruction on how to install this?  As I am a total noob.  I am using an btt sir mini version 2 from an old Ender 5 I had.

**Reply by 3DCoded** [reply]
Currently, West3D is working on a video series with the 3MS. You can find it on the homepage of the documentation.

**Reply by ENF3471** [reply]
That is what drove me to the 3ms.  I was thinking of doing the chameleon but thought this would be easier as I had most of the parts so out of pocket would be minimal for experimenting.  I have the board all set and connected to the K1 max.   I was trying to follow the sortware instructions but was getting a bit lost.    One thing was is do I log into the machine through putty or use the cincel to install the firmware on to the SKR mini?

**Reply by 3DCoded** [reply]
You log into the machine through Putty to run all shell commands.

**Reply by ENF3471** [reply]
Thanks I will try that using the information above.  You are correct even though you root the k1 the original Creality print still works.  If I run a print from the machine

**Reply by ENF3471** [reply]
not sure is this is happening due to fact this is a K1 and its not true Klipper even after rooting. but I get this when putting in the first code of the instructions.

![Screenshot 2024-10-29 091051](https://github.com/user-attachments/assets/15c5d2e8-9547-4795-8c5b-dc26914e588e)

**Reply by ENF3471** [reply]
sorry for my not understanding of this as I have installed Octoprint in the past as well as rooted my K1.  thought this should be a walk in the park as the instructions are fairly easy to understand but it seems that is not the case for me.

**Reply by 3DCoded** [reply]
This is likely because the K1 is using a heavily modified Klipper and a reduced Linux system. The 3MS is a lot easier to install on standard Klipper setups. For the K1, can you try running the following for the `cd ~/klipper`:

```
cd /usr/share/klipper
```

**Reply by ENF3471** [reply]
that seemed to work.  But it wont make the menuconfig file.  I can create a make file but not the menuconfig.
![Screenshot 2024-10-29 091051](https://github.com/user-attachments/assets/c8136f68-d7d2-4c8c-bd69-b948753be862)

**Reply by 3DCoded** [reply]
That's an issue with Creality's custom Klipper. Running `make menuconfig` is fairly common on most Klipper setups, even without a 3MS. If you can get your K1 to run a standard Klipper setup, installation will go a lot smoother.

**Reply by 3DCoded** [reply]
The 3MS documentation for the firmware is a simplified version of the [official Klipper docs](https://www.klipper3d.org/Installation.html#building-and-flashing-the-micro-controller).

**Comment by ENF3471** [comment]
I am not finding anyway to update it to full klipper.  It seems that pivlbig may have had success with this.  Maybe he will comment on the steps he took to get it working.

**Reply by pvilbig** [reply]
Sorry! Meant to write thanking you for helping because I do have it working! I’m working on fine tuning the distances right now. But the thread here that we were figuring it out on helped. I did a fresh root and reinstalled everything and it worked flawlessly with Klipper screen. Life got a little crazy sorry! I know I’m late but thank you for all your help. I will try to help when I can but I did start a new job but I am free on weekends so it might take me a little bit to get back to you guys!

PS so glad I got a notification about the interview video. Glad to see your getting recognized for this awesome MMS!!!

**Reply by ENF3471** [reply]
thanks for writing back.  I am glad you got it working on your K1.   When you have sometime I would love to hear how you got it installed.   As I am having issues with just the simplest of tasks due to Creality print.  as you can see above I got into the root but I cannot even get menuconfig to work

Thanks for any help in advanced

**Reply by 3DCoded** [reply]
The 3MS now works with Happy Hare v3 (beta). As far as I understand, HH does support Creality K1 series 3D printers. Check out the beta in #29.

**Comment by brentdeacon23** [comment]
Anyone had any luck flashing klipper from the k1? I'm lost at this point

**Reply by pvilbig** [reply]
Same here had to redo the setup so now HH won't even install

**Reply by 3DCoded** [reply]
What do you mean Happy Hare won't even install?

**Reply by pvilbig** [reply]
Okay well I got that going but now I have this issue:

{"code": "key341", "msg": "Section 'menu __main __mmu' is not a valid config section\n\nOnce the underlying issue is corrected, use the \"RESTART\"\ncommand to reload the config and restart the host software.\nPrinter is halted\n", "values": ["menu __main __mmu"]}

**Comment by pvilbig** [comment]
Any luck flashing Happy-Hare to the K1 fam?

**Reply by pvilbig** [reply]
So this error I am getting when I try to flash the mcu's
Error opening serial /dev/ttyS7 with error read failed: device reports readiness to read but returned no data (device disconnected or multiple access on port?)

**Reply by 3DCoded** [reply]
You should be using a /dev/serial/by-id path

**Reply by pvilbig** [reply]
Ok so I got it all going except klipperscreen. I’m now getting this error

![IMG_0926](https://github.com/user-attachments/assets/85a874ec-1549-463d-a8fa-b3de8e2f6401)

**Reply by 3DCoded** [reply]
> Ok so I got it all going except klipperscreen. I’m now getting this error
> 
> 
> 
> ![IMG_0926](https://github.com/user-attachments/assets/85a874ec-1549-463d-a8fa-b3de8e2f6401)
> 
> 

I would recommend you ask this question in the Happy Hare Discord.

**Comment by 3DCoded** [comment]
@pvilbig @brentdeacon23 ,

Here's a guide to install proper Klipper onto K1 series printers, and to flash boards.
https://github.com/lamtranBKHN/creality_k1_happy_hare

**Reply by brentdeacon23** [reply]
Omg this is exactly what I need, now I just need some reliable micro sd cards, installed klipper via kiauh today and now I can’t even get it to recognise the card in the pc

**Reply by pvilbig** [reply]
Wow! I think I wound up finding that right before this! Yes perfect guide!
